oauth2-server/src/main/java/com/clouway/oauth2/ClientAuthorizationActivity.java

```diff
@@ -49,7 +49,7 @@ public Response execute(String identityId, Request request) {
     }
 
     Set<String> scopes = Sets.newTreeSet(Splitter.on(" ").omitEmptyStrings().split(scope));
-    Optional<Authorization> possibleAuthorizationResponse = clientAuthorizationRepository.authorize(client, identityId, scopes, responseType);
+    Optional<Authorization> possibleAuthorizationResponse = clientAuthorizationRepository.authorize(client, identity.idenityId, identity.scopes, responseType);
```

Comment:
>  The authorization server MAY fully or partially ignore the scope
   requested by the client, based on the authorization server policy or
   the resource owner's instructions.  If the issued access token scope
   is different from the one requested by the client, the authorization
   server MUST include the "scope" response parameter to inform the
   client of the actual scope granted.

But as this is library, it should be decided by the client code which to be used and how, not by the library. The library should pass what was requested.